compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1787,8 +1819,39 @@ impl<'a> Parser<'a> {
                 Ok(true) => {}
                 Ok(false) => unreachable!(),
                 Err(mut err) => {
+                    // Qualifier keywords ordering check
+                    // Extern is not checked since it is the last expected one
+                    let misplaced_qualifier =
+                        [kw::Const, kw::Async, kw::Unsafe].iter().any(|&k| self.check_keyword(k));
+
+                    if misplaced_qualifier {
+                        // This will allow the machine fix to directly place the keyword in the correct place
+                        let current_qual_sp = (if self.check_keyword(kw::Const) {
```

Comment:
Consider making this `Option<_>` where `None` would be returned from the `else` branch that currently has `unreachable!()` in it.